File: 1910. Remove All Occurrences of a Substring/1910. Remove All Occurrences of a Substring.js
/**
 * @param {string} s
 * @param {string} part
 * @return {string}
 */
var removeOccurrences = function(s, part) {
  let stack = []
  for (let char of s) {
    stack.push(char)
    if (stack.length>=part.length){
        const isPart = stack.slice(-part.length).join("")
        if(isPart===part){
            for(let _ =0 ; _<part.length;_++){
                stack.pop()
            }
        }
    }
  } 
  return stack.join("")
};

File: 1910. Remove All Occurrences of a Substring/1910. Remove All Occurrences of a Substring.py
class Solution:
    def removeOccurrences(self, s: str, part: str) -> str:
        stack = []
        for char in s:
            stack.append(char)
            if len(stack)>= len(part) and "".join(stack[-len(part):])==part:
                for _ in range(len(part)):
                    stack.pop()
        return "".join(stack)

sol = Solution()
print(sol.removeOccurrences(s = "daabcbaabcbc", part = "abc"))
print(sol.removeOccurrences(s = "axxxxyyyyb", part = "xy"))


